### ISSUE TITLE
Print out a report of times.

### ISSUE THREAD

**Issue by Ben-PH** [sagiegurari/cargo-make#446]
**Features Description**
The outputs of the commands being run are independent of the overall build script. As a result, the extra commands (such as running `time cargo make $TASK`) or tasks are required to get information on the the build script as a whole. This feature would provide an option that would have something like the following printed at the end.

```text
=== Time Stats ===
Total: xm y.yys
TaskA: 40% (xm y.yys)
TaskC: 23% (xm y.yys)
TaskE: 9%  (xm y.yys)
Others: 28% (xm y.yys, 4 tasks omitted)
```

**Describe the solution you'd like**
```toml
[tasks.build]
description = "Build client and server"
# as root-task: enable final-time report when running this task
# as dependency of a task with root-task enabling it also: is included
# in the breakdown of the final time
time = true 
aggregate_times = true
workspace = false
dependencies = ["build_client", "build_server"]

[tasks.build_client]
description = "Build client"
workspace = false
time = false
install_crate = { crate_name = "wasm-pack", binary = "wasm-pack", test_arg = "-V" }
command = "wasm-pack"
args = ["build", "client", "--target", "web", "--out-name", "package", "--dev"]

[tasks.build_server]
description = "Build server"
workspace = false
time = true
command = "cargo"
args = ["build", "--package", "server"]
```
running the above example with `cargo make build`, an output might look like this

```toml
=== Time Stats ===
Total: 10m 0.00s

Server: 65% (6m 30.0s)
Other: 35% (3m 30.0s, 1 omited)
```

**Comment by sagiegurari** [comment]
Interesting idea, i will check it out. few notes however,
* sometimes your build will be splited to multiple processes (wor example when working in a workspace, each member is a sub process) so you will have multiple summaries.
* it would be turned off by default. you will have to provide some cli flag or value in the makefile config section to turn it on. many people complain it is printing too much as it is, so trying to gate any extra printing.

**Comment by sagiegurari** [comment]
@Ben-PH I just pushed to the development branch the time summary.
The printout looks as follows:
```console
[cargo-make] INFO - =====Time Summary=====
[cargo-make] INFO - install-rustfmt:             0.00%     0.01 seconds
[cargo-make] INFO - format:                      0.63%     1.44 seconds
[cargo-make] INFO - format-flow:                 0.00%     0.00 seconds
[cargo-make] INFO - build:                       8.44%     19.37 seconds
[cargo-make] INFO - test:                        19.65%    45.11 seconds
[cargo-make] INFO - test-thread-safe:            19.66%    45.12 seconds
[cargo-make] INFO - test-single-threaded:        15.98%    36.67 seconds
[cargo-make] INFO - test-multi-phases-cleanup:   0.00%     0.00 seconds
[cargo-make] INFO - test-multi-phases-flow:      35.65%    81.84 seconds
[cargo-make] INFO - test-flow:                   0.00%     0.00 seconds
[cargo-make] INFO - default:                     0.00%     0.00 seconds
```

To make it work you will need to either pass the new --time-summary flag or add to the makefile config section the time_summary=true flag.
its automatically turned on in CI env.

Any feedback is welcome 😃

**Comment by sagiegurari** [comment]
this is now released in the new cargo-make version 0.32.2
